bundles/org.openhab.binding.wizlighting/src/main/java/org/openhab/binding/wizlighting/internal/handler/WizLightingMediatorImpl.java

```diff
@@ -0,0 +1,252 @@
+    /**
+     * Inits the mediator WizBulbUpdateReceiverRunnable thread. This thread is
+     * responsible to receive all packets from Wiz Bulbs, and redirect the messages
+     * to mediator.
+     */
+    private void initMediatorWizBulbUpdateReceiverRunnable() {
+        WizLightingUpdateReceiverRunnable receiver = this.receiver;
+        Thread receiverThread = this.receiverThread;
+        // try with handler port if is null
+        if ((receiver == null)
+                || ((receiverThread != null) && (receiverThread.isInterrupted() || !receiverThread.isAlive()))) {
+            try {
+                logger.trace("Receiver thread is either null, interrupted, or dead.");
+                WizLightingUpdateReceiverRunnable newReceiver = new WizLightingUpdateReceiverRunnable(this,
+                        WizLightingBindingConstants.DEFAULT_LISTENER_UDP_PORT);
+                Thread newThread = new Thread(newReceiver);
+                newThread.start();
+                this.receiver = newReceiver;
+                this.receiverThread = newThread;
+                logger.trace("A new receiver thread has been started.");
+            } catch (SocketException e) {
+                logger.debug("Cannot start the socket with default port {}...", e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Returns all the {@link Thing} registered.
+     *
+     * @returns all the {@link Thing}.
+     */
+    @Override
+    public Set<Thing> getAllThingsRegistered() {
+        return this.handlersRegisteredByThing.keySet();
+    }
+
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.STATIC)
+    public void setNetworkAddressService(NetworkAddressService networkAddressService) {
+        this.networkAddressService = networkAddressService;
+        logger.trace("Network Address Service has been set in the mediator.");
+    }
+
+    public void unsetNetworkAddressService(NetworkAddressService networkAddressService) {
+        this.networkAddressService = null;
+        logger.trace("Network Address Service has been unset from the mediator.");
+    }
+
+    private String getMyIpAddress() {
+        String myIpAddress = networkAddressService.getPrimaryIpv4HostAddress();
+        if (myIpAddress == null) {
+            logger.warn("Network interface did not return an IP address!");
+            return "OHIPAddress";
+        }
+        logger.info("IP of OpenHab device is {}.", myIpAddress);
+        return myIpAddress;
+    }
+
+    private String getMyMacAddress() {
+        String myMacAddress;
+        try {
+            myMacAddress = NetworkUtils.getMyMacAddress();
+            if (myMacAddress == null) {
+                logger.warn("No network interface could be found.  Mac of OpenHab device is unknown.");
+                return "OHMACAddress";
+            }
+        } catch (Exception e) {
+            logger.warn("Mac Address of OpenHab device is invalid.");
+            return "OHMACAddress";
```

Comment:
Why are you returning an invalid mac address?

Reply:
Because it's convenient to have the address already defined and then check if it's valid when actually needed rather than throwing in null checks everywhere.

Reply:
Why not use `Optional<String>` instead?

Reply:
Because it's not really optional.  The bulbs won't accept a "registration" (to receive updates) without being told the (valid) mac address of the receiver.